engine/select_members.py

```diff
@@ -10,146 +10,161 @@
 import logging
 import os
 import sys
-from collections import Counter
 from datetime import datetime
 
 import click
-import numpy as np
 
 from engine.tolerance import tolerance
 from util.click_util import cli_help
 from util.dataframe_ops import test_stats_file_with_tolerances
 from util.log_handler import logger
 
 
-# finds members and a corresponding tolerance factor validating for all stats files.
 def find_members_and_factor_validating_for_all_stats_files(
     random_tolerance_file_name,
     stats_file_name,
     member_type,
-    min_member_num,
-    max_member_num,
-    total_member_num,
+    max_member_count,
+    total_member_count,
     min_factor,
     max_factor,
-    iterations,
 ):  # pylint: disable=too-many-positional-arguments
+    """
+    Find a minimal subset of ensemble members and a tolerance factor such that
+    all statistics files are validated.
+
+    Given a set of ensemble members, the algorithm attempts to identify a subset
+    that satisfies specified tolerances.
+    Starting with each ensemble member individually, the algorithm checks how
+    many of the remaining ensemble members fall within the defined tolerances
+    when compared to the current member.
+    The ensemble member that results in the largest number of other members
+    satisfying the tolerances is selected as the reference.
+
+    This step is repeated, progressively validating additional members, until
+    all ensemble members are validated or a predefined maximum number of
+    iterations is reached. If the latter occurs, the tolerance factor is
+    increased and the process restarts.
```

Comment:
Is this actually true? Is there still a predefined maximum number of iterations? Doesn't it go until `max_member_count` and `max_factor` are reached?

Reply:
Every iteration one member is added, so the `max_iteration` and `max_member_count` is the same. But I agree this can be clarified here.